hera_cal/abscal_funcs.py

```diff
@@ -585,6 +585,90 @@ def delay_slope_lincal(model, data, antpos, wgts=None, df=9.765625e4, medfilt=Tr
+
+    median_over_freq(V_ij,xy^data / V_ij,xy^model) = dot(Phi_x, r_i) - dot(Phi_y, r_j)
+
+    Parameters:
+    -----------
+    model : visibility data of refence model, type=DataContainer
+            keys are antenna-pair + polarization tuples, Ex. (1, 2, 'xx').
+            values are complex ndarray visibilities.
+            these must 2D arrays, with [0] axis indexing time
+            and [1] axis indexing frequency.
+
+    data : visibility data of measurements, type=DataContainer
+           keys are antenna pair + pol tuples (must match model), values are
+           complex ndarray visibilities matching shape of model
+
+    antpos : type=dictionary, antpos dictionary. antenna num as key, position vector as value.
+
+    wgts : weights of data, type=DataContainer, [default=None]
+           keys are antenna pair + pol tuples (must match model), values are real floats
+           matching shape of model and data. These are only used to find delays from
+           itegrations that are unflagged for at least two frequency bins. In this case, 
+           the delays are assumed to have equal weight, otherwise the delays take zero weight.
+
+    verbose : print output, type=boolean, [default=False]
+
+    tol : type=float, baseline match tolerance in units of baseline vectors (e.g. meters)
+
+    Output:
+    -------
+    fit : dictionary containing frequency-indpendent phase slope, e.g. Phi_ns_x 
+          for each position component and polarization [radians / meter].
+    """
+    echo("...configuring linsolve data for global_phase_slope_logcal", verbose=verbose)
+
+    # get keys from model and data dictionaries
+    keys = sorted(set(model.keys()) & set(data.keys()))
+    ants = np.unique(antpos.keys())
+
+    # make weights if None and make flags
+    if wgts is None:
+        wgts = odict()
+        for i, k in enumerate(keys):
+            wgts[k] = np.ones_like(data[k], dtype=np.float)
+    flags = DataContainer({k: wgts[k]==0 for k in keys})
+
+    # average data over baselines
+    reds = redcal.get_reds(antpos, bl_error_tol=tol, pols=data.pols(), low_hi=True)
+    avg_data, avg_flags, red_keys = avg_data_across_red_bls(DataContainer({k: data[k] for k in keys}), 
+                                    antpos, flags=flags, broadcast_flags=False, tol=tol, reds=reds)
+    avg_model, _, _ = avg_data_across_red_bls(DataContainer({k: model[k] for k in keys}), 
+                      antpos, flags=flags, broadcast_flags = False, tol=tol, reds=reds)
+    
+    # build linear system
+    ls_data, ls_wgts = {}, {}
+    for rk in red_keys:
+        # build equation string 
+        eqn_str = '{}*Phi_ew_{} + {}*Phi_ns_{} - {}*Phi_ew_{} - {}*Phi_ns_{}'
+        eqn_str = eqn_str.format(antpos[rk[0]][0], rk[2][0], antpos[rk[0]][1], rk[2][0],
+                                 antpos[rk[1]][0], rk[2][1], antpos[rk[1]][1], rk[2][1])
+
+        # calculated frequency median of unflagged angle(data/model)
```

Comment:
is it better to average the first data across freq, and then take `angle`?

Reply:
so what about phase wrapping issues... i.e. when the ratio of data / model isn't that close and oscillates between and around 0 and -2pi?

Reply:
It's a fair question. Both methods suffer from phase wraps potentially... the hope is that the median fixes it either way as long as it doesn't phase wrap too much.

This is just my experience with one data file, which admittedly, is picked to be very similar to the calibrator. I think it's worth revisiting this if the fix doesn't stop sources from moving around in our images.

Also, the point of this is to just get things closer... the iterative TT will get us the rest of the way if there's lots of phase wrapping.